client/src/pages/role-details/RoleDetails.tsx

```diff
@@ -0,0 +1,144 @@
+import React, { useEffect, useState } from 'react';
+import {
+  Button,
+  Card,
+  CardContent,
+  Container,
+  FormControl,
+  Grid,
+  InputLabel,
+  MenuItem,
+  Select,
+  TextField,
+  Typography,
+  Alert,
+  Checkbox,
+} from '@mui/material';
+import { useParams } from 'react-router-dom';
+import { Controller, useForm } from 'react-hook-form';
+import { useMutation, useQuery, useQueryClient } from 'react-query';
+import { getRole, getFeatures } from '../../data';
+import PageHeader from '../../components/page-header/PageHeader';
+import Spacer from '../../components/spacer/Spacer';
+import SaveBar from '../../components/save-bar/save-bar';
+
+export default function RoleDetails() {
+  const { id } = useParams();
+  if (!id) {
+    return (
+      <Container>
+        <Alert severity="error">Role ID is not defined</Alert>
+      </Container>
+    );
+  }
+  // getFeatures().then(x => {
+
+  // })
+  // console.log(features)
+
+  const queryClient = useQueryClient();
+  
+  let features = useQuery('users', getFeatures)
```

Comment:
`'users'`? Do you mean `features`?

Also, I strongly recommend that you wrap the useQuery as a hook called `useAppFeatures` and have that in the data files as well. This way people can ensure they are using the correct query cache